app/models/step_by_step_page.rb

```diff
@@ -83,6 +83,10 @@ def can_be_edited?
     !scheduled_for_publishing?
   end
 
+  def steps_have_content?
+    steps.any? && steps.map(&:contents).all?(&:present?)
+  end
```

Comment:
Can the movement of this code be fixed up to the 54db8c2c994add0711568d3530ab1bf5434679ef commit?

Reply:
Do you mean the movement of `steps.any?`

Reply:
I mean the movement of this method (`steps_have_content?`)

Reply:
So I made `steps_have_content?` public at the moment that I needed to access it outside of the class. If it's hard to see from the diff why the method has moved, I can update the commit message.

Reply:
Ah ok, makes sense 👍